### PR TITLE
fix(policy): enhance proto validation across policy requests

### LINKED ISSUE #1640: Policy: KAS registration lacks validation of URI
### Background

Registering a KAS to a platform currently requires a unique `URI` and a `public_key` that is either `remote` or `cached`: https://github.com/opentdf/platform/blob/05512477604d894b9eba3dca137d9960608abf40/service/policy/kasregistry/key_access_server_registry.proto#L22

Soon, it will also include (optionally) a common name for the KAS which is also unique: https://github.com/opentdf/platform/issues/153

However, the `uri` field in the proto above has no validation at present. 

```shell
./otdfctl policy kasr create -r https://remote.com/kas -u 'b' --json
{
  "id": "87859c4d-b7f2-4366-8900-658b2a97de49",
  "uri": "b", // this should not be valid
  "public_key": {
    "PublicKey": {
      "Remote": "https://remote.com/kas"
    }
  },
  "metadata": {
    "created_at": {
      "seconds": 1728938163,
      "nanos": 43877000
    },
    "updated_at": {
      "seconds": 1728938163,
      "nanos": 43877000
    }
  }
}
```

### Acceptance Criteria

1. define allowed protocol schemes for KAS URIs
2. enforce valid URI format with a CEL expression at the proto level

### PR DESCRIPTION
**Pull request by jakedoublev**
Resolves #1640 and a host of similar issues with validation throughout policy service protos

### CODE REVIEW

**Comment by jakedoublev** [comment]
As part of this work I explored custom validators, which would look like:

```proto
// validator.proto
edition = "2023";

package policy;

import "buf/validate/validate.proto";

extend buf.validate.StringRules {
  bool optional_uuid = 70000001 [            // opens up custom validator '.string.(optional_uuid)'
    (buf.validate.predefined).cel = {
      id: "uuid_format",
      message: "field must be a valid UUID",
      expression: "size(this) == 0 || this.matches('[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}')"
    }
  ];
}
```

Then in any proto we have an optional UUID, we could do:
```proto
message ListKeyAccessServerGrantsRequest {
  // Optional
  string kas_id = 1 [(buf.validate.field).string.(optional_uuid) = true]; // custom validator
}
```

The custom validator feature updates we need were [merged into protovalidate](https://github.com/bufbuild/protovalidate/pull/246) about a month ago and aren't supported by all our plugins yet ([buf.build/grpc-ecosystem/gateway](https://buf.build/grpc-ecosystem/gateway?version=v2.22.0) at the latest (v2.22.0) doesn't yet support "editions" which is a requirement :disappointed: ). Not only that, but it's also likely plugins for other languages won't support them either, so it's too bleeding edge for us.

We still improve validation of many fields in this PR, but it requires duplicate rules and CEL expressions we cannot yet unify with custom validations.

File: service/policy/attributes/attributes_test.go [hunk 1]
```diff
@@ -187,6 +187,86 @@ func TestCreateAttribute_ValueInvalid_Fails(t *testing.T) {
...
+	}
+
+	for _, test := range bad {
+		invalidAttrKAS := &attributes.AttributeKeyAccessServer{
+			AttributeId:       test.attrID,
+			KeyAccessServerId: test.kasID,
+		}
+		err := getValidator().Validate(invalidAttrKAS)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "string.uuid")
```
[the remark this answers is not in the document]
Comment (ryanulit) [reply service/policy/attributes/attributes_test.go:227]: Also, I'm fairly certain I wrote a bunch of tests that don't make these error contains checks so maybe we can add them in this PR too?

**Comment by ryanulit** [comment]
LGTM depending on your thoughts on my comments.  Won't approve yet due to code freeze.

File: service/policy/kasregistry/key_access_server_registry.proto [hunk 2]
```diff
@@ -19,11 +22,19 @@ message ListKeyAccessServersResponse {
...
 
+// TODO: optional validation below should be through a custom validator, which is too bleeding edge at present without full plugin support
+
 message CreateKeyAccessServerRequest {
   // Required
-  string uri = 1 [(buf.validate.field).required = true];
+  string uri = 1 [(buf.validate.field).cel = {
+      id: "uri_format",
+      message: "URI must be a valid URL (e.g., 'https://demo.com/') followed by additional segments. Each segment must start and end with an alphanumeric character, can contain hyphens, alphanumeric characters, and slashes.",
+      expression: "this.matches('^https?://[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?(\\\\.[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?)*(:[0-9]+)?(/.*)?$')"
```
Comment (jakedoublev) [inline service/policy/kasregistry/key_access_server_registry.proto:32]: @dmihalcik-virtru Hey Dave, I noticed we weren't doing validation on this field previously and some of our test pipelines were creating KASes with URIs that didn't use TLS. Does updating the URI format for the registered KASes throughout to `http` make sense to you?

File: service/policy/kasregistry/key_access_server_registry.proto [hunk 3]
```diff
@@ -65,16 +88,29 @@ message KeyAccessServerGrants {
...
+  }];
+  // Optional
+  // Filter LIST by URI of a registered Key Access Server.
   // If neither is provided, grants from all registered KASs to policy attribute objects are returned.
-  string kas_id = 1;
-  string kas_uri = 2;
+  string kas_uri = 2 [(buf.validate.field).cel = {
+      id: "optional_uri_format",
+      message: "Optional URI must be a valid URL (e.g., 'https://demo.com/') followed by additional segments. Each segment must start and end with an alphanumeric character, can contain hyphens, alphanumeric characters, and slashes.",
+      expression: "size(this) == 0 || this.matches('^https?://[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?(\\\\.[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?)*(:[0-9]+)?(/.*)?$')"
```
Comment (jakedoublev) [inline service/policy/kasregistry/key_access_server_registry.proto:112]:
```suggestion
      expression: "size(this) == 0 || this.isUri()"
```

File: service/policy/kasregistry/key_access_server_registry.proto [hunk 4]
```diff
@@ -33,10 +44,19 @@ message CreateKeyAccessServerResponse {
...
-  string id = 1 [(buf.validate.field).required = true];
-  string uri = 2;
+  string id = 1 [
+    (buf.validate.field).string.uuid = true
+  ];
+  // Required
+  string uri = 2 [(buf.validate.field).cel = {
+      id: "optional_uri_format",
+      message: "Optional URI must be a valid URL (e.g., 'https://demo.com/') followed by additional segments. Each segment must start and end with an alphanumeric character, can contain hyphens, alphanumeric characters, and slashes.",
+      expression: "size(this) == 0 || this.matches('^https?://[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?(\\\\.[a-zA-Z0-9]([a-zA-Z0-9\\\\-]{0,61}[a-zA-Z0-9])?)*(:[0-9]+)?(/.*)?$')"
```
Comment (jakedoublev) [inline service/policy/kasregistry/key_access_server_registry.proto:54]:
```suggestion
      expression: "size(this) == 0 ||  this.isUri()"
"
```

File: service/policy/kasregistry/key_access_server_registry.proto (hunk 2 above)
Comment (jakedoublev) [inline service/policy/kasregistry/key_access_server_registry.proto:32]:
```suggestion
      expression: "this.isUri()"
```

Comment (jakedoublev) [reply service/policy/kasregistry/key_access_server_registry.proto:32]: We will allow TLS as optional per existing standards in the OpenTDF Platform

**Pull request merged**